_docs-sources/pipelines/architecture/index.md

```diff
@@ -0,0 +1,45 @@
+# Architecture
+
+## Overview
+
+There are three major components in the Gruntwork Pipelines architecture:
+
+1. The orchestrator
+2. The dispatcher
+3. The executor
+
+### The orchestrator
+
+The orchestrator identifies each [infrastructure change](../core-concepts/#infrastructure-change) in the Pull Request or git commit, classifies the type of infrastructure change it is (e.g. NewAwsAccount vs. OpenTfModuleUpdate) and determines the right [pipelines actions](../core-concepts/#pipelines-actions) to run based on that infrastructure change. For each infrastructure change, it calls the dispatcher.
+
+### The dispatcher
+
+The dispatcher takes as input an infrastructure change and a list of which pipelines actions to run with along with what parameters, and calls the executor to run these actions on the right infrastructure change.
+
+### The executor
+
+The executor executes an ordered list of pipelines actions against a specific pull request or git commit in response to a specific infrastructure. It then posts the status and output of these pipelines actions to the pull request, or takes other actions to ensure that the user has visibility into what happened.
+
+## Execution Flow
+
+You can see these components in action below:
+
+![Gruntwork Pipelines Execution Flow](/img/pipelines/how-it-works/pipelines_execution_flow.png)
+
```

Comment:
The diagram is helpful in quickly understanding how it all works. My main feedback is around using consistent terminology and embracing Pipelines Actions as a first-class concept (over mere terragrunt plan/apply/destroy).

In that spirit, could we change:
- "Calculate infra-change-set from changes" to "Parse infra-change-set from changes"
- "For each infra change in infra change set" to "For each infra-change in infra-change set"
- "Determine change type" to "Determine infra-change type"
- "Determine terragrunt command" to "Determine Pipelines Actions"
- "Run terragrunt plan/apply/destroy" to "Run Pipelines Actions

Reply:
Can do